shared/src/main/java/me/neznamy/tab/shared/features/nametags/NameTag.java

```diff
@@ -354,8 +357,7 @@ public void onLoginPacket(TabPlayer player) {
     @Override
     public void onVanishStatusChange(@NotNull TabPlayer player) {
         for (TabPlayer viewer : TAB.getInstance().getOnlinePlayers()) {
-            if (viewer.hasPermission(TabConstants.Permission.SEE_VANISHED)) continue;
-            if (player.isVanished()) {
+            if (player.isVanished() && (!viewer.hasPermission(TabConstants.Permission.SEE_VANISHED) && (premiumVanish == null || !premiumVanish.canSee(viewer, player)))) {
```

Comment:
Are you sure this will work properly? If a player vanishes, the the will double register their team for players who can see vanished players.